src/Sarif/Writers/SarifLogger.cs

```diff
@@ -447,12 +447,16 @@ private void CaptureArtifact(ArtifactLocation fileLocation)
                 catch (ArgumentException) { } // Unrecognized encoding name
             }
 
+            HashData hashData = null;
+            AnalysisTargetToHashDataMap?.TryGetValue(fileLocation.Uri.LocalPath, out hashData);
```

Comment:
>AnalysisTargetToHashDataMap?.TryGetValue [](http://example.com/codeflow?start=12&length=40)

if we have analysisTarget, we will check if we have the hashdata. If yes, we will use it

Reply:
Your comment describes the literal code change but you haven't described why the code change has been made.